src/features/canvass/components/LocationDialog/pages/HouseholdsVisitPage.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Step,
+  StepButton,
+  StepContent,
+  Stepper,
+  ToggleButton,
+  ToggleButtonGroup,
+  Typography,
+} from '@mui/material';
+import { FC, useEffect, useState } from 'react';
+
+import { ZetkinLocation, ZetkinMetric } from 'features/areaAssignments/types';
+import PageBase from './PageBase';
+import { Msg, useMessages } from 'core/i18n';
+import messageIds from 'features/canvass/l10n/messageIds';
+import { MetricResponse } from 'features/canvass/types';
+
+type HouseholdVisitPageProps = {
+  location: ZetkinLocation;
+  metrics: ZetkinMetric[];
+  onBack: () => void;
+  onLogVisit: (metrics: MetricResponse[]) => void;
+  selectedHouseholsdIds: number[];
+};
+
+const HouseholdVisitPage: FC<HouseholdVisitPageProps> = ({
+  selectedHouseholsdIds,
+  location,
+  metrics,
+  onBack,
+  onLogVisit,
+}) => {
+  const messages = useMessages(messageIds);
+  const [responseByMetricId, setResponseByMetricId] = useState<
+    Record<number, MetricResponse>
+  >({});
+  const [step, setStep] = useState(0);
+  const [loading, setLoading] = useState(false);
+  location;
+
+  useEffect(() => {
+    setResponseByMetricId({});
+    setStep(0);
+  }, []);
+
+  return (
+    <PageBase
+      actions={
+        step >= metrics.length && (
+          <Button
+            disabled={loading}
+            fullWidth
+            onClick={async () => {
+              setLoading(true);
+              const responses = Object.values(responseByMetricId).map(
+                (response) => response
+              );
+
+              const filteredResponses = responses.filter(
+                (response) => !!response.response
+              );
+              await onLogVisit(filteredResponses);
+              setLoading(false);
+            }}
+            startIcon={
+              loading ? (
+                <CircularProgress color="secondary" size="20px" />
+              ) : null
+            }
+            variant="contained"
+          >
+            {`Submit report for  ${selectedHouseholsdIds.length} households`}
+          </Button>
+        )
+      }
+      onBack={onBack}
+      title={`Selected ${selectedHouseholsdIds.length} households`}
+    >
+      <Stepper activeStep={step} orientation="vertical">
```

Comment:
Am I right to think that this is a copy of the same UI for a single household? If so, maybe it needs to be refactored (can happen after this PR) to also accommodate the changes from #2965?

Reply:
Good idea! I will do it in a separate task as you suggest